src/garage/experiment/meta_evaluator.py

```diff
@@ -22,40 +24,61 @@ class MetaEvaluator:
         n_exploration_traj (int): Number of trajectories to gather from the
             exploration policy before requesting the meta algorithm to produce
             an adapted policy.
+        n_test_rollouts (int): Number of rollouts to use for each adapted
+            policy. The adapted policy should forget previous rollouts when
+            `.reset()` is called.
         prefix (str): Prefix to use when logging. Defaults to MetaTest. For
             example, this results in logging the key 'MetaTest/SuccessRate'.
             If not set to `MetaTest`, it should probably be set to `MetaTrain`.
+        test_task_names (list[str]): List of task names to test. Should be in
+            an order consistent with the `task_id` env_info, if that is
+            present.
 
     """
 
     # pylint: disable=too-few-public-methods
 
     def __init__(self,
-                 runner,
                  *,
                  test_task_sampler,
                  max_path_length,
-                 n_test_tasks=None,
-                 n_exploration_traj=1,
-                 prefix='MetaTest'):
+                 n_exploration_traj=10,
+                 n_test_tasks=1,
+                 n_test_rollouts=1,
+                 prefix='MetaTest',
+                 test_task_names=None):
         self._test_task_sampler = test_task_sampler
         if n_test_tasks is None:
             n_test_tasks = test_task_sampler.n_tasks
         self._n_test_tasks = n_test_tasks
+        self._n_test_rollouts = n_test_rollouts
         self._n_exploration_traj = n_exploration_traj
-        self._test_sampler = runner.make_sampler(
-            LocalSampler, n_workers=1, max_path_length=max_path_length)
+        self._max_path_length = max_path_length
         self._eval_itr = 0
         self._prefix = prefix
+        self._test_task_names = test_task_names
+        self._test_sampler = None
 
-    def evaluate(self, algo):
+    def evaluate(self, algo, test_rollouts_per_task=None):
         """Evaluate the Meta-RL algorithm on the test tasks.
 
         Args:
             algo (garage.np.algos.MetaRLAlgorithm): The algorithm to evaluate.
+            test_rollouts_per_task (int or None): Number of rollouts per task.
 
         """
+        if test_rollouts_per_task is None:
+            test_rollouts_per_task = self._n_test_rollouts
         adapted_trajectories = []
+        logger.log('Sampling for adapation and meta-testing...')
+        if self._test_sampler is None:
+            self._test_sampler = LocalSampler.from_worker_factory(
+                WorkerFactory(seed=get_seed(),
+                              max_path_length=self._max_path_length,
+                              n_workers=1,
+                              worker_class=DefaultWorker),
+                agents=algo.get_exploration_policy(),
+                envs=self._test_task_sampler.sample(1))
```

Comment:
Should it be 
```
...
envs=self._test_task_sampler.sample(self._n_test_tasks))
```
?

Reply:
Nope. The meta evaluator goes through one task at a time. See the next line of code.

Reply:
Since in `meta_squash` branch it is, just want to make sure it's the correct implementation here

Reply:
Hmm. I'm fairly certain that part of the `meta_squash` branch is wrong, but thanks for checking.